src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java

```diff
@@ -48,21 +65,102 @@ public String getFeedSourceId () {
+            // editor) in order to see the results.
+            // FIXME need to add snapshot namespace...
+            for (DbTransformation transformation : rules.getActiveTransformations(feedVersion, DbTransformation.class)) {
+                addNextJob(new ArbitraryTransformJob(owner, snapshot.id, transformation));
+            }
+            // If the user has selected to create a new version from the resulting snapshot, do so here.
+            if (rules.createNewVersion) {
+                addNextJob(new CreateFeedVersionFromSnapshotJob(feedSource, snapshot, owner));
+            }
+        }
+
+        // FIXME: Should we overwrite the input GTFS dataset if transforming in place?
+
         // TODO: Any other activities that need to be run (e.g., module-specific activities).
+        if (DataManager.isModuleEnabled("deployment")) {
+//            Project project = feedSource.retrieveProject();
+//            if (project.autoDeployId)
+            // TODO: Get deployment, update feed version for feed source, and kick off deployment to server that
+            //  deployment is currently pointed at.
+        }
     }
 
     @Override
     public void jobFinished () {
         if (!status.error) {
             // Note: storing a new feed version in database is handled at completion of the ValidateFeedJob subtask.
             status.completeSuccessfully("New version saved.");
+            // OK, let's map out the different transformation possibilities:
+            // 1. We only have zip transformations. These can either apply directly to the imported zip (dangerous)
+            // or they can apply to a cloned version of the import. In that case, the retrieval method should be set to
+            // VERSION_CLONE (otherwise, can be set to any other combination of retrieval methods).
+            // 2. We only have db transformations. These will always apply to a cloned snapshot of a feed version to
+            // avoid the primary db namespace having different contents from the zip file. Possibilities here are that
+            // the snapshot can end up hanging (i.e., not published) or once all of the transformations have been
+            // applied, we can create a new feed version from the snapshot (retrieval method being produced in house??).
+            // 3. Let's say we have zip transformations and db transformations.
+            // 3a: Zip transformations apply to imported version, db transformations apply to snapshot or get published (no big deal)
+            // 3b: Zip transformations apply to cloned version, db transformations apply to snapshot or get published.
+            //  ---- This case is a little trickier, because we could end up with two cloned_versions. Let's say we want
+            //  a SQL query to run on the original data, create a new GTFS file, and then replace a GTFS+ file from some
+            //  other version. FTR1 = clone
```

Comment:
This whole block of comments confused me. A lot of it doesn't seem to necessarily apply to the immediate section of code below.

Also, what does `FTR1 = clone` mean?

Reply:
Sorry this code block was for me when I was figuring out how to structure everything. I will move to javadoc for jobLogic and clarify.